src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,216 @@
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths) => new SpanND<T>(address, lengths, false);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new SpanND<T>(address, lengths, false, strides);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths) => new ReadOnlySpanND<T>(address, lengths, false);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new ReadOnlySpanND<T>(address, lengths, false, strides);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        public static Tensor<T> CreateFromEnumerable<T>(IEnumerable<T> data)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            T[] values = data.ToArray();
+            return new Tensor<T>(values, [values.Length], false);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        public static Tensor<T> CreateUniform<T>(params ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>, IFloatingPoint<T>
+        {
+            var linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            T[] values = new T[linearLength];
+            Random rand = new Random();
+            for (int i = 0; i < values.Length; i++)
+                values[i] = T.CreateChecked(rand.NextSingle());
```

Comment:
We should likely be using `Random.Shared` instead of allocating a new random and we should probably consider if this is the right shape for the API.

This is going to be under distributed for large floating-point (`double`, `decimal`, or even future types like `binary128`) and due to how floating-point works won't necessarily be a uniform distribution for `half` (or `float` if we changed this to `rand.NextDouble()`).

-- The general consideration is that IEEE 754 floating-point isn't itself uniformly distributed, its distributed on an exponential curve where you get `2^n` entries every power of 2. We currently get a linear distribution by doing `1.0 / 2^n` and then scaling by a random number between 0 and 2^n. Where `n=53` for `double` and `n=24` for `float`. However, because of the non-linear distribution for the type in general and because of `round to even`, we end up getting bias (and therefore non-uniform distribution) if we take a uniform `double` sample and downcast it to `float`.

We could potentially provide explicit APIs for the built-in/common floating-point types for convenience. But I think the better approach might be some general `CreateAndFill<T, TSample>` API, where we then provide some default `sample` algorithms for things like `Uniform` and `GuassianNormal`